### PR TITLE
WIP: Exchange tests

### PR DESCRIPTION
**Pull request by Elendan**

### CODE REVIEW

File: test/NosCore.Tests/HandlerTests/ExchangeControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+            _session.InitializeAccount(account1);
+            _session2.InitializeAccount(account2);
+            _session3.InitializeAccount(account3);
+            
+            _itemBuilderService = new ItemBuilderService(items, new List<IHandler<Item, Tuple<IItemInstance, UseItemPacket>>>());
+            _handler = new ExchangePacketController(_worldConfiguration, _exchangeService);
+            _handler2 = new ExchangePacketController(_worldConfiguration, _exchangeService);
+            _handler3 = new ExchangePacketController(_worldConfiguration, _exchangeService);
+            _map = new MapInstance(new Map
```
Comment (erwan-joly) [inline test/NosCore.Tests/HandlerTests/ExchangeControllerTests.cs]: don't copy past things you don't need

File: test/NosCore.Tests/HandlerTests/ExchangeControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+            {
+                CharacterId = 1,
+                Name = "TestExistingCharacter",
+                Slot = 1,
+                AccountId = account1.AccountId,
+                MapId = 1,
+                State = CharacterState.Active
+            };
+
+            var account2 = new AccountDto { Name = "Account2", Password = "test".ToSha512() };
```
Comment (erwan-joly) [inline test/NosCore.Tests/HandlerTests/ExchangeControllerTests.cs]: Account will need a language

File: test/NosCore.Tests/HandlerTests/ExchangeControllerTests.cs (hunk 1 above)
Comment (Elendan) [reply test/NosCore.Tests/HandlerTests/ExchangeControllerTests.cs]: I do need a mapinstance, users should not be able to exchange if they are on different maps

**Comment by Elendan** [comment]
Will be done later, too many changes have been ported to the master branch

**Pull request closed without merging**